File: java/src/main/java/org/dilithia/sdk/model/MultisigTx.java
package org.dilithia.sdk.model;

import java.util.List;
import java.util.Map;

/**
 * Represents a pending multisig transaction.
 *
 * @param txId      the unique transaction identifier
 * @param contract  the target contract for the proposed call
 * @param method    the target method for the proposed call
 * @param args      the call arguments
 * @param approvals the addresses that have approved this transaction
 */
public record MultisigTx(
        String txId,
        String contract,
        String method,
        Map<String, Object> args,
        List<String> approvals
) {}

File: dotnet/Dilithia.Sdk/Models/MultisigTx.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// A pending multisig transaction.
/// </summary>
/// <param name="TxId">The unique transaction identifier.</param>
/// <param name="Contract">The target contract for the proposed call.</param>
/// <param name="Method">The target method for the proposed call.</param>
/// <param name="Approvals">The addresses that have approved this transaction.</param>
public record MultisigTx(string TxId, string Contract, string Method, List<string> Approvals);
